Spigot-API-Patches/0243-Added-SoundBuilder.patch

```diff
@@ -0,0 +1,349 @@
++        this.receivers = null;
++        return this;
++    }
++
++    /**
++     * @param receivers List of players to receive this sound, or null for all players in the
++     * world
++     * @return a reference to this object.
++     * @deprecated The use of list's is not appreciated
++     */
++    @NotNull
++    @Deprecated
++    public SoundBuilder receivers(@Nullable List<Player> receivers) {
++        this.receivers = receivers != null ? Lists.newArrayList(receivers) : null;
++        return this;
++    }
++
++    /**
++     * @param receivers List of players to receive this sound, or null for all players in the
++     * world
++     * @return a reference to this object.
++     */
++    @NotNull
++    public SoundBuilder receivers(@Nullable Collection<Player> receivers) {
++        this.receivers = receivers != null ? Lists.newArrayList(receivers) : null;
++        return this;
++    }
++
++    /**
++     * @param receivers List of players to be receive this sound, or null for all players in the
++     * world
++     * @return a reference to this object.
++     */
++    @NotNull
++    public SoundBuilder receivers(@Nullable Player... receivers) {
++        this.receivers = receivers != null ? Lists.newArrayList(receivers) : null;
++        return this;
++    }
++
++    /**
++     * Selects all players within a cuboid selection around the sound location, within the
++     * specified bounding box. If you want a more spherical check, see {@link #receivers(int,
++     * boolean)}
++     *
++     * @param radius amount to add on all axis
++     * @return a reference to this object.
++     */
++    @NotNull
++    public SoundBuilder receivers(int radius) {
++        return receivers(radius, radius);
++    }
++
++    /**
++     * Selects all players within the specified radius around the sound location. If byDistance is
++     * false, behavior uses cuboid selection the same as {@link #receivers(int, int)} If byDistance is
++     * true, radius is tested by distance in a spherical shape
++     *
++     * @param radius amount to add on each axis
++     * @param byDistance true to use a spherical radius, false to use a cuboid
++     * @return a reference to this object.
++     */
++    @NotNull
++    public SoundBuilder receivers(int radius, boolean byDistance) {
++        if (!byDistance) {
++            return receivers(radius, radius, radius);
++        } else {
++            this.receivers = Lists.newArrayList();
++            for (Player nearbyPlayer : location.getWorld()
++                .getNearbyPlayers(location, radius, radius, radius)) {
++                Location loc = nearbyPlayer.getLocation();
++                double x = NumberConversions.square(location.getX() - loc.getX());
++                double y = NumberConversions.square(location.getY() - loc.getY());
++                double z = NumberConversions.square(location.getZ() - loc.getZ());
++                if (Math.sqrt(x + y + z) > radius) {
```

Comment:
x+y+z > radiusSquared (compute the squared version outside of the loop)

Reply:
I was gonna suggest just using loc.distance(loc2) but then itll be inconsistent with the x/z version below i guess, so im fine wit hthis.